### ISSUE TITLE
OIDC backchannel, multiple sessions on same client, only one backchannel request

### ISSUE THREAD

**Issue by Jayashree-Rajendran** [keycloak/keycloak#27359]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

authentication

### Describe the bug

In backchannel logout of our OIDC client implementation, I noticed the following behavior in Keycloak when multiple sessions exist.

Steps :
Log in the user in the same client using, e.g., two browsers. 
You will see two sessions for the registered OIDC client in Keycloak
On the user’s sessions tab, click on “Logout all sessions”.

This results in one single backchannel request being triggered from keycloak
where in logout_token I can identify the user through SID (ID of one of the two open sessions) or SUB (internal ID of the keycloak user).

Now you can imagine that the client itself may have internal session information (session cookie) that need to be invalidate/terminate as a result of the backchannel request

However, it is impossible for the client to identify which of the sessions to actively terminate, as seemingly only one of the active sessions in Keycloak will be submitted as a backchannel logout request.

How can client identify which all sessions to terminate? 

### Version

23.0.7

### Regression

- [ ] The issue is a regression

### Expected behavior

I would have expected one request per session to get both SID logout requests in the client

### Actual behavior

Keycloak sending sid of one of the sessions

### How to Reproduce?

Log in the user in the same client using, e.g., two browsers. 
You will see two sessions for the registered OIDC client in Keycloak
On the user’s sessions tab, click on “Logout all sessions”.

**Comment by imazzeii** [comment]
https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken

_A Logout Token MUST contain either a sub or a sid Claim, and MAY contain both. If a sid Claim is not present, the intent is that all sessions at the RP for the End-User identified by the iss and sub Claims be logged out._ 

so in this case logout token should not contain any sid claim; the application should keep track of sid, sub and sessionid cookie associations

**Comment by NobodysNightmare** [comment]
I agree to @imazzeii 's reading of the OIDC Backchannel Logout specification:

* If only a single session should end, the `sid` must be included in the logout token
* If all sessions should end, the `sid` **must be absent** from the logout token

Right now Keycloak seems to be able to configure per client whether the client needs a `sid` or not in Logout tokens, but it does not seem to distinguish between ending a single session or many sessions.

**Comment by itsvse** [comment]
Encountering the same issue, unable to exit all clients sessions of this user, hope to fix the bug.
